### ISSUE TITLE
EnpointHandler checkPath logic is broken

### ISSUE THREAD

**Issue by andr3h3nriqu3s11** [zigzap/zap#136]
Currently, if you create an endpoint handler and set checkPath to false the endpoint code will never be called.

```zig
        var fooHandler = zap.Middleware.EndpointHandler(Handler, Context).init(
            fooEndpoint.endpoint(),
            null,
            .{
                .checkPath = false,
            },
        );
```

This seems to be caused by the logic in `EndpointHandler.onRequest`.
As `self.endpoint.onRequest(r)` is only called inside this if, and if checkPath is false it will always skip endpoint, instead of always running the enpoint.

```zig
            if (self.options.checkPath and
                std.mem.startsWith(u8, r.path orelse "", self.endpoint.settings.path))
            {
                self.endpoint.onRequest(r);
            }
```

Reverting [9a80747](https://github.com/zigzap/zap/commit/9a80747cdd66e204b9223d806f54c87acd05cb7b) should fix the issue.

**Comment by renerocksai** [comment]
Correct! Oops. The referenced commit is wrong but I know what to do.

**Comment by renerocksai** [comment]
fixed in ae5c9278335d8e1133cd6d22707323dda712e120

**Comment by andr3h3nriqu3s11** [comment]
Oops! Copied the parent commit... Thank you for the fix!
